wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -143,43 +144,79 @@ class KotlinGenerator private constructor(
   }
 
   /**
+   * Generates all [TypeSpec]s for the given [Service].
+   *
    * If [onlyRpc] isn't null, this will generate code only for this onlyRpc; otherwise, all RPCs of
    * the [service] will be code generated.
    */
-  fun generateService(service: Service, onlyRpc: Rpc? = null): TypeSpec {
+  fun generateServiceTypeSpecs(service: Service, onlyRpc: Rpc? = null): List<TypeSpec> {
+    val typeSpecs = mutableListOf<TypeSpec>()
+    if (rpcRole == RpcRole.SERVER) {
+      typeSpecs.add(generateService(service, onlyRpc))
+    } else {
+      typeSpecs.add(generateService(service, onlyRpc, generateClientImplementation = false))
+      typeSpecs.add(generateService(service, onlyRpc, generateClientImplementation = true))
+    }
+    return typeSpecs
+  }
+
+  private fun generateService(
+    service: Service,
+    onlyRpc: Rpc?,
+    generateClientImplementation: Boolean
+  ): TypeSpec {
     val serviceName = generatedServiceName(service, onlyRpc)
-    val builder = if (rpcRole == RpcRole.SERVER) {
+    val builder = if (rpcRole == RpcRole.SERVER || !generateClientImplementation) {
       TypeSpec.interfaceBuilder(serviceName)
+          .addSuperinterface(com.squareup.wire.Service::class.java)
     } else {
-      TypeSpec.classBuilder(serviceName)
+      val serviceImplementationName = ClassName(
+          serviceName.packageName,
+          "Real${serviceName.simpleName}"
```

Comment:
I thought we were going with "Grpc" as a prefix.

Reply:
Ah, true, I missed that comment. Submitting a PR shortly.